llvm/include/llvm/MC/CAS/MCCASObjectV1.h

```diff
@@ -6,6 +6,99 @@
 //
 //===----------------------------------------------------------------------===//
 
+/// \verbatim
+/// CAS Schema:
+///
+/// Debug Line Section
+///
+/// DebugLineSectionRef: The DebugLineSectionRef represents the debug_line
+/// section, it contains the relocations for that section as well as any addends
+/// and their offsets and sizes, that were zero-ed out for deduplication
+/// purposes.
+///
+/// DebugLineRef: The DebugLineRef represents one function's contribution to the
+/// debug_line section, or one line table. Each function will have it's own line
+/// table entry in the debug_line section.
+///
+/// ┌─────────────────┐
+/// │  DebugLineRef   │
+/// │  [Line Table 1] │
+/// │                 │         ┌──────────────────────┐
+/// └──────────▲──────┘         │  DebugLineSectionRef │
+///       .    ├────────────────┤                      │
+///       .    │                │     [Relocations]    │
+///       .    │                │     [Addends]        │
+/// ┌──────────▼──────┐         └──────────────────────┘
+/// │  DebugLineRef   │
+/// │  [Line Table N] │
+/// │                 │
+/// └─────────────────┘
+///
+/// Debug String Section
+///
+/// DebugStringSectionRef: The DebugStringSectionRef represents the debug_str
+/// section.
+///
+/// DebugStrRef: A DebugStrRef represents one string in the debug_str section,
+/// each string gets its own cas object.
+///                         ┌────────────────┐
+///            ┌────────────┤  DebugString   ├───────────┐
+///            │            │  SectionRef    │           │
+///            │            └────────────────┘           │
+///     ┌──────▼───────┐                         ┌───────▼──────┐
+///     | DebugStrRef  │                         │ DebugStrRef  |
+///     │ [String 1]   │          ...            │ [String N]   │
+///     └──────────────┘                         └──────────────┘
+///
+///
+/// Debug Info Section
+///
+/// DebugInfoSectionRef: The DebugInfoSectionRef represents the debug_line
+/// section, it contains the relocations for that section as well as any addends
+/// and their offsets and sizes, that were zero-ed out for deduplication
```

Comment:
Is this true? Isn't that data stored in the distinctdataref?

Reply:
No, we had an experimental patch where we moved the dw_form_addrs into the distinctdataref, but since I could not address some of the comments that were left in that patch, it has not been merged yet.

https://github.com/apple/llvm-project/pull/5343